metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java

```diff
@@ -86,7 +86,31 @@
     @Column(nullable = false)
     private Type type;
 
-    File() {}
+    /**
+     * Release date control: get the *earliest* release date from all studies which link to this file.
+     */
+    @Formula("(SELECT min(study.release_date) FROM file " +
+            "INNER JOIN analysis_files on file.id = analysis_files.files_id " +
+            "INNER JOIN analysis on analysis_files.analysis_id = analysis.id " +
+            "INNER JOIN study on analysis.study_id = study.id " +
+            "WHERE file.id=id)")
+    @JsonIgnore
+    private LocalDate releaseDate;
+
+    /**
+     * Get the studyIds.
+     */
+
+    @Formula("(SELECT string_agg(concat(study.accession,'.',study.version),',') FROM file " +
+            "INNER JOIN analysis_files on file.id = analysis_files.files_id " +
+            "INNER JOIN analysis on analysis_files.analysis_id = analysis.id " +
+            "INNER JOIN study on analysis.study_id = study.id " +
+            "WHERE file.id=id)")
```

Comment:
I would very much suggest to move the constant part of the expression (starting from `FROM` until the end of expression) into a string constant. So that the formula looks like something like this:

```java
@Formula("(SELECT string_agg(concat(study.accession,'.',study.version),',')" + fileQueryExpression + ")")
```

And also do the same for all other objects. It would _greatly_ reduce the possibility of errors when updating some of it in the future.